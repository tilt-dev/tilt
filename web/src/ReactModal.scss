@import "constants";

.ReactModal__Overlay {
  z-index: $z-modal;
  display: flex;
  align-items: center;
  justify-content: center;
}