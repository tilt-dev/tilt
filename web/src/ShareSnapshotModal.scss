@import "constants";
@import "ReactModal";

.ReactModal__Content.ShareSnapshotModal {
  width: 800px;
}
.ShareSnapshotModal-title {
  margin: 0;
  font-family: $font-sans-serif;
  font-size: $font-size-small;
  text-transform: uppercase;
  background-color: $color-gray-light;
  color: $color-white;
  padding-top: $spacing-unit / 2;
  padding-bottom: $spacing-unit / 2;
  padding-left: $spacing-unit;
  padding-right: $spacing-unit;
}
.ShareSnapshotModal-description {
  font-family: $font-sans-serif;
  line-height: 1.75;
  margin-top: $spacing-unit * 1.5;
  margin-left: $spacing-unit * 3;
  // Get the line break where I want it:
  margin-right: $spacing-unit * 4;
}

// Learn about Snapshots
.ShareSnapshotModal-intro {
  display: flex;
  margin-top: $spacing-unit * 1.5;
  margin-left: $spacing-unit * 2;
  margin-right: $spacing-unit * 2;
}
.ShareSnapshotModal-introImage {
  width: 200px;
}
.ShareSnapshotModal-details {
 margin-left: $spacing-unit * 1.5;
}
.ShareSnapshotModal-detailsList {
  list-style: none;
  position: relative; // Anchor the list bullets
}
.ShareSnapshotModal-detailsList > li {
  margin-bottom: $spacing-unit / 3;
}
.ShareSnapshotModal-detailsList > li::before {
  content: "";
  width: $spacing-unit;
  height: $spacing-unit;
  position: absolute;
  left: -$spacing-unit;
  background-image:url('data:image/svg+xml;utf8,<svg fill="none" height="8" viewBox="0 0 6 8" width="6" xmlns="http://www.w3.org/2000/svg"><path d="m6 4-5.25 3.4641v-6.928202z" fill="%23c4c4c4"/></svg>');
  background-repeat: no-repeat;
  background-position: center;
}
.ShareSnapshotModal-docsLink {
  display: inline-block;
  text-decoration: none;
  background-color: $color-off-white;
  margin-top: $spacing-unit / 4;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
  line-height: 1.75;
}
.ShareSnapshotModal-docsLink:hover {
  text-decoration: underline;
}
.ShareSnapshotModal-getStarted {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  border-top: 1px dotted $color-gray-lightest;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
  // This visually centers the button, whose shadow throws things off a bit:
  padding-top: $spacing-unit * 0.6;
  padding-bottom: $spacing-unit * 0.8;
}


// Create a Snapshot Link
.ShareSnapshotModal-shareLinkWrap {
  background-color: $color-off-white;
  margin-top: $spacing-unit;
  padding-top: $spacing-unit * 1.5;
  padding-bottom: $spacing-unit * 0.8;
  padding-left: $spacing-unit * 2;
  padding-right: $spacing-unit * 2;
}
.ShareSnapshotModal-shareLink {
  display: flex;
}
.ShareSnapshotModal-shareLink-urlBox {
  background-color: $color-white;
  border: 1px solid $color-gray-lightest;
  padding-top: $spacing-unit / 6;
  padding-bottom: $spacing-unit / 6;
  padding-left: $spacing-unit / 3;
  padding-right: $spacing-unit / 3;
  box-sizing: border-box;
  display: inline-block;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
}
.ShareSnapshotModal-shareLink-url {
  user-select: all;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.ShareSnapshotModal-shareLink-placeholder {
  color: $color-gray-lightest;
}
.ShareSnapshotModal-shareLinkInfo {
  display: flex;
  justify-content: space-between;
  margin-top: $spacing-unit * 0.3;
}

// Tilt Cloud Info
.ShareSnapshotModal-manageSnapshots {
  font-family: $font-sans-serif;
  font-size: $font-size-small;
  margin-top: $spacing-unit;
  margin-bottom: $spacing-unit;
  margin-left: $spacing-unit * 3;
  margin-right: $spacing-unit * 3;
}
.ShareSnapshotModal-tiltCloudLink {
  text-decoration: none;
}
.ShareSnapshotModal-tiltCloudLink span {
  border-bottom: 2px solid $color-gray;
  margin-right: $spacing-unit / 4;
}
.ShareSnapshotModal-tiltCloudLink svg {
  fill: $color-gray-light;
}
.ShareSnapshotModal-tiltCloudLink:hover span {
  border-bottom-color: $color-blue;
}
.ShareSnapshotModal-tiltCloudLink:hover svg {
  fill: $color-gray-darkest;
}

// Button Styles
.ShareSnapshotModal-button {
  cursor: pointer;
  font-family: $font-sans-serif;
  text-transform: uppercase;
  color: $color-white;
  border: 0;
  padding-top: $spacing-unit / 3;
  padding-bottom: $spacing-unit / 3;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
  text-decoration: none;
  transition: color 300ms;
  display: flex;
  justify-content: center;
  align-items: center;
}
.ShareSnapshotModal-button--inline {
  font-size: $font-size-smallest;
  background-color: $color-gray-dark;
  min-width: 95px; // Don't let button text wrap
  box-sizing: border-box;
}
.ShareSnapshotModal-button--inline:hover {
  color: $color-blue;
}
.ShareSnapshotModal-button--cta {
  font-size: $font-size-small;
  background-color: $color-blue;
  box-shadow: -5px 5px 0 0 $color-gray;
  margin-left: $spacing-unit;
  transition-property: box-shadow;
  transition-duration: 0.3s;
  transition-timing-function: linear;
}
.ShareSnapshotModal-button--cta:hover {
  box-shadow: -3px 3px 0 0 $color-gray;
}
.ShareSnapshotModal-button--cta:active {
  box-shadow: -1px 1px 0 0 $color-gray;
}
