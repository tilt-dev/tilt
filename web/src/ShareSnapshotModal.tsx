import React, { PureComponent } from "react"
import Modal from "react-modal"
import "./ShareSnapshotModal.scss"
import cookies from "js-cookie"
import intro from "./assets/png/share-snapshot-intro.png"
import { ReactComponent as ArrowSvg } from "./assets/svg/arrow.svg"

type props = {
  handleSendSnapshot: () => void
  handleClose: () => void
  snapshotUrl: string
  tiltCloudUsername: string | null
  tiltCloudSchemeHost: string
  tiltCloudTeamID: string | null
  isOpen: boolean
  highlightedLines: number | null
}

export default class ShareSnapshotModal extends PureComponent<props> {
  render() {
    return (
      <Modal
        onRequestClose={this.props.handleClose}
        isOpen={this.props.isOpen}
        className="ShareSnapshotModal"
      >
        <h2 className="ShareSnapshotModal-title">Share a Shapshot</h2>
        <section className="ShareSnapshotModal-pane u-flexColumn">
          <p className="ShareSnapshotModal-description">
            Get a link to a{" "}
            {this.props.tiltCloudTeamID ? "private team " : null} snapshot â€” a
            browsable, sharable view of the current state of your Tilt session.
          </p>
          {this.renderCallToAction()}
        </section>
        {this.maybeRenderManageSnapshots()}
      </Modal>
    )
  }

  renderCallToAction() {
    if (this.props.tiltCloudUsername) {
      return (
        <section className="ShareSnapshotModal-shareLinkWrap">
          {this.renderShareLink()}
          {this.renderShareLinkInfo()}
        </section>
      )
    }

    return (
      <section>
        {this.renderIntro()}
        {this.renderGetStarted()}
      </section>
    )
  }

  renderIntro() {
    return (
      <div className="ShareSnapshotModal-intro">
        <div className="u-inlineBlock">
          <img
            src={intro}
            className="ShareSnapshotModal-introImage"
            alt="hand holding up a copy of the Tilt User Interface"
          />
        </div>
        <div className="u-inlineBlock ShareSnapshotModal-details">
          <ul className="ShareSnapshotModal-detailsList">
            <li>Share errors easily</li>
            <li>Explore logs in-context</li>
            <li>Work together to figure out the problem</li>
          </ul>
          <a
            className="ShareSnapshotModal-docsLink"
            href="https://docs.tilt.dev/snapshots.html"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn more in our docs
          </a>
        </div>
      </div>
    )
  }

  renderGetStarted() {
    return (
      <div className="ShareSnapshotModal-getStarted">
        <p className="u-inlineBlock">
          Link Tilt to Tilt Cloud (just takes a minute)
        </p>
        <form
          action={this.props.tiltCloudSchemeHost + "/start_register_token"}
          target="_blank"
          method="POST"
          onSubmit={ShareSnapshotModal.notifyTiltOfRegistration}
        >
          <input name="token" type="hidden" value={cookies.get("Tilt-Token")} />
          <input
            type="submit"
            className="ShareSnapshotModal-button ShareSnapshotModal-button--cta"
            value="Get Started"
          />
        </form>
      </div>
    )
  }

  renderShareLink() {
    const hasLink = this.props.snapshotUrl !== ""
    return (
      <div className="ShareSnapshotModal-shareLink">
        <div className="ShareSnapshotModal-shareLink-urlBox">
          {hasLink ? (
            <p className="ShareSnapshotModal-shareLink-url">
              {this.props.snapshotUrl}
            </p>
          ) : (
            <p className="ShareSnapshotModal-shareLink-placeholder">
              No Link Generated Yet
            </p>
          )}
        </div>
        {this.renderGetLinkButton()}
      </div>
    )
  }

  renderGetLinkButton() {
    const hasLink = this.props.snapshotUrl !== ""
    if (hasLink) {
      return (
        <a
          className="ShareSnapshotModal-button ShareSnapshotModal-button--inline"
          href={this.props.snapshotUrl}
          target="_blank"
          rel="noopener noreferrer"
        >
          Open
        </a>
      )
    }
    return (
      <button
        className="ShareSnapshotModal-button ShareSnapshotModal-button--inline"
        onClick={this.props.handleSendSnapshot}
      >
        Get Link
      </button>
    )
  }

  renderShareLinkInfo() {
    let lines = this.props.highlightedLines
    return (
      <section className="ShareSnapshotModal-shareLinkInfo">
        <p className="ShareSnapshotModal-loginState">
          Sharing as <strong>{this.props.tiltCloudUsername}</strong>
        </p>
        {lines && (
          <p>
            {lines} Line{lines > 1 ? "s" : ""} Highlighted
          </p>
        )}
      </section>
    )
  }

  maybeRenderManageSnapshots() {
    if (!this.props.tiltCloudUsername) {
      return null
    }
    return (
      <section className="ShareSnapshotModal-manageSnapshots">
        <p>
          Manage your snapshots on{" "}
          <a
            className="ShareSnapshotModal-tiltCloudLink"
            href={this.props.tiltCloudSchemeHost + "/snapshots"}
            target="_blank"
            rel="noopener noreferrer"
          >
            <span>Tilt Cloud</span>
            <ArrowSvg />
          </a>
        </p>
        {this.props.tiltCloudTeamID ? (
          <p>
            View snapshots from your{" "}
            <a
              className="ShareSnapshotModal-tiltCloudLink"
              rel="noopener noreferrer"
              href={`${this.props.tiltCloudSchemeHost}/team/${this.props.tiltCloudTeamID}/snapshots`}
              target="_blank"
            >
              <span>team</span>
              <ArrowSvg />
            </a>
          </p>
        ) : null}
      </section>
    )
  }

  static notifyTiltOfRegistration() {
    let url = `${window.location.protocol}//${window.location.host}/api/user_started_tilt_cloud_registration`
    fetch(url, {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
    })
  }
}
