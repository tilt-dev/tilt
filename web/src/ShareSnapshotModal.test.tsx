import React from "react"
import ReactDOM from "react-dom"
import ReactModal from "react-modal"
import renderer from "react-test-renderer"
import ShareSnapshotModal from "./ShareSnapshotModal"

const fakeSendsnapshot = () => {}
const fakeHandleCloseModal = () => {}
let originalCreatePortal = ReactDOM.createPortal

describe("ShareSnapshotModal", () => {
  beforeEach(() => {
    ReactModal.setAppElement(document.body)
    let mock: any = (node: any) => node
    ReactDOM.createPortal = mock
  })

  afterEach(() => {
    ReactDOM.createPortal = originalCreatePortal
  })

  it("renders with modal open w/o known username", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          isOpen={true}
          snapshotUrl="http://test.com"
          tiltCloudUsername={null}
          tiltCloudSchemeHost={"https://cloud.tilt.dev"}
          tiltCloudTeamID={null}
          highlightedLines={null}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders with modal open w/ known username", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          isOpen={true}
          snapshotUrl="http://test.com"
          tiltCloudUsername={"tacocat"}
          tiltCloudSchemeHost={"https://cloud.tilt.dev"}
          tiltCloudTeamID={null}
          highlightedLines={null}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders without snapshotUrl", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          isOpen={true}
          snapshotUrl=""
          tiltCloudUsername={null}
          tiltCloudSchemeHost={"https://cloud.tilt.dev"}
          tiltCloudTeamID={null}
          highlightedLines={null}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders with modal closed", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          isOpen={false}
          snapshotUrl="http://test.com"
          tiltCloudUsername={null}
          tiltCloudSchemeHost={"https://cloud.tilt.dev"}
          tiltCloudTeamID={null}
          highlightedLines={null}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders team link", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          isOpen={true}
          snapshotUrl="http://test.com"
          tiltCloudUsername={"Hello"}
          tiltCloudSchemeHost={"https://cloud.tilt.dev"}
          tiltCloudTeamID={"abcdefg"}
          highlightedLines={null}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
