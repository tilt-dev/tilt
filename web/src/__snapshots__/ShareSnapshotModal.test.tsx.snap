// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareSnapshotModal renders team link 1`] = `
<div
  className="ReactModal__Overlay ReactModal__Overlay--after-open"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="ReactModal__Content ReactModal__Content--after-open ShareSnapshotModal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="ShareSnapshotModal-title"
    >
      Share a Shapshot
    </h2>
    <section
      className="ShareSnapshotModal-pane u-flexColumn"
    >
      <p
        className="ShareSnapshotModal-description"
      >
        Get a link to a
         
        private team 
         snapshot — a browsable, sharable view of the current state of your Tilt session.
      </p>
      <section
        className="ShareSnapshotModal-shareLinkWrap"
      >
        <div
          className="ShareSnapshotModal-shareLink"
        >
          <div
            className="ShareSnapshotModal-shareLink-urlBox"
          >
            <p
              className="ShareSnapshotModal-shareLink-url"
            >
              http://test.com
            </p>
          </div>
          <a
            className="ShareSnapshotModal-button ShareSnapshotModal-button--inline"
            href="http://test.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Open
          </a>
        </div>
        <section
          className="ShareSnapshotModal-shareLinkInfo"
        >
          <p
            className="ShareSnapshotModal-loginState"
          >
            Sharing as 
            <strong>
              Hello
            </strong>
          </p>
        </section>
      </section>
    </section>
    <section
      className="ShareSnapshotModal-manageSnapshots"
    >
      <p>
        Manage your snapshots on
         
        <a
          className="ShareSnapshotModal-tiltCloudLink"
          href="https://cloud.tilt.dev/snapshots"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            Tilt Cloud
          </span>
          <svg>
            arrow.svg
          </svg>
        </a>
      </p>
      <p>
        View snapshots from your
         
        <a
          className="ShareSnapshotModal-tiltCloudLink"
          href="https://cloud.tilt.dev/team/abcdefg/snapshots"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            team
          </span>
          <svg>
            arrow.svg
          </svg>
        </a>
      </p>
    </section>
  </div>
</div>
`;

exports[`ShareSnapshotModal renders with modal closed 1`] = `null`;

exports[`ShareSnapshotModal renders with modal open w/ known username 1`] = `
<div
  className="ReactModal__Overlay ReactModal__Overlay--after-open"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="ReactModal__Content ReactModal__Content--after-open ShareSnapshotModal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="ShareSnapshotModal-title"
    >
      Share a Shapshot
    </h2>
    <section
      className="ShareSnapshotModal-pane u-flexColumn"
    >
      <p
        className="ShareSnapshotModal-description"
      >
        Get a link to a
         
         snapshot — a browsable, sharable view of the current state of your Tilt session.
      </p>
      <section
        className="ShareSnapshotModal-shareLinkWrap"
      >
        <div
          className="ShareSnapshotModal-shareLink"
        >
          <div
            className="ShareSnapshotModal-shareLink-urlBox"
          >
            <p
              className="ShareSnapshotModal-shareLink-url"
            >
              http://test.com
            </p>
          </div>
          <a
            className="ShareSnapshotModal-button ShareSnapshotModal-button--inline"
            href="http://test.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Open
          </a>
        </div>
        <section
          className="ShareSnapshotModal-shareLinkInfo"
        >
          <p
            className="ShareSnapshotModal-loginState"
          >
            Sharing as 
            <strong>
              tacocat
            </strong>
          </p>
        </section>
      </section>
    </section>
    <section
      className="ShareSnapshotModal-manageSnapshots"
    >
      <p>
        Manage your snapshots on
         
        <a
          className="ShareSnapshotModal-tiltCloudLink"
          href="https://cloud.tilt.dev/snapshots"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            Tilt Cloud
          </span>
          <svg>
            arrow.svg
          </svg>
        </a>
      </p>
    </section>
  </div>
</div>
`;

exports[`ShareSnapshotModal renders with modal open w/o known username 1`] = `
<div
  className="ReactModal__Overlay ReactModal__Overlay--after-open"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="ReactModal__Content ReactModal__Content--after-open ShareSnapshotModal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="ShareSnapshotModal-title"
    >
      Share a Shapshot
    </h2>
    <section
      className="ShareSnapshotModal-pane u-flexColumn"
    >
      <p
        className="ShareSnapshotModal-description"
      >
        Get a link to a
         
         snapshot — a browsable, sharable view of the current state of your Tilt session.
      </p>
      <section>
        <div
          className="ShareSnapshotModal-intro"
        >
          <div
            className="u-inlineBlock"
          >
            <img
              alt="hand holding up a copy of the Tilt User Interface"
              className="ShareSnapshotModal-introImage"
              src="share-snapshot-intro.png"
            />
          </div>
          <div
            className="u-inlineBlock ShareSnapshotModal-details"
          >
            <ul
              className="ShareSnapshotModal-detailsList"
            >
              <li>
                Share errors easily
              </li>
              <li>
                Explore logs in-context
              </li>
              <li>
                Work together to figure out the problem
              </li>
            </ul>
            <a
              className="ShareSnapshotModal-docsLink"
              href="https://docs.tilt.dev/snapshots.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn more in our docs
            </a>
          </div>
        </div>
        <div
          className="ShareSnapshotModal-getStarted"
        >
          <p
            className="u-inlineBlock"
          >
            Link Tilt to Tilt Cloud (just takes a minute)
          </p>
          <form
            action="https://cloud.tilt.dev/start_register_token"
            method="POST"
            onSubmit={[Function]}
            target="_blank"
          >
            <input
              name="token"
              type="hidden"
            />
            <input
              className="ShareSnapshotModal-button ShareSnapshotModal-button--cta"
              type="submit"
              value="Get Started"
            />
          </form>
        </div>
      </section>
    </section>
  </div>
</div>
`;

exports[`ShareSnapshotModal renders without snapshotUrl 1`] = `
<div
  className="ReactModal__Overlay ReactModal__Overlay--after-open"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="ReactModal__Content ReactModal__Content--after-open ShareSnapshotModal"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <h2
      className="ShareSnapshotModal-title"
    >
      Share a Shapshot
    </h2>
    <section
      className="ShareSnapshotModal-pane u-flexColumn"
    >
      <p
        className="ShareSnapshotModal-description"
      >
        Get a link to a
         
         snapshot — a browsable, sharable view of the current state of your Tilt session.
      </p>
      <section>
        <div
          className="ShareSnapshotModal-intro"
        >
          <div
            className="u-inlineBlock"
          >
            <img
              alt="hand holding up a copy of the Tilt User Interface"
              className="ShareSnapshotModal-introImage"
              src="share-snapshot-intro.png"
            />
          </div>
          <div
            className="u-inlineBlock ShareSnapshotModal-details"
          >
            <ul
              className="ShareSnapshotModal-detailsList"
            >
              <li>
                Share errors easily
              </li>
              <li>
                Explore logs in-context
              </li>
              <li>
                Work together to figure out the problem
              </li>
            </ul>
            <a
              className="ShareSnapshotModal-docsLink"
              href="https://docs.tilt.dev/snapshots.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn more in our docs
            </a>
          </div>
        </div>
        <div
          className="ShareSnapshotModal-getStarted"
        >
          <p
            className="u-inlineBlock"
          >
            Link Tilt to Tilt Cloud (just takes a minute)
          </p>
          <form
            action="https://cloud.tilt.dev/start_register_token"
            method="POST"
            onSubmit={[Function]}
            target="_blank"
          >
            <input
              name="token"
              type="hidden"
            />
            <input
              className="ShareSnapshotModal-button ShareSnapshotModal-button--cta"
              type="submit"
              value="Get Started"
            />
          </form>
        </div>
      </section>
    </section>
  </div>
</div>
`;
